File: mcap-web/src/main/resources/static/js/service/function/sysResourceService.js
/**
 * author:Marshal
 * description:service层,与数据库交互
 */
app.service("sysResourceService",function($http){
    //base crud
    this.query=function(pageNum,pageSize,condition){
        return $http.post("../sys/resource/query?pageNum="+pageNum+"&pageSize="+pageSize,condition);
    }

    this.save=function (sysResource) {
        return $http.post("../sys/resource/save",sysResource);
    }

    this.delete=function (selectedIds) {
        return $http.get("../sys/resource/delete?selectedIds="+selectedIds);
    }

    this.queryById=function (id) {
        return $http.get("../sys/resource/queryById?id="+id);
    }

    //other opertions
    this.getResourceOptions=function(){
        return $http.get("../sys/resource/getResourceOptions");
    }
});

File: halcyon-system/src/main/java/com/marshal/halcyon/system/service/SysResourceService.java
package com.marshal.halcyon.system.service;


import com.marshal.halcyon.system.entity.SysResource;

import java.util.List;
import java.util.Map;

public interface SysResourceService {
    List<SysResource> select(SysResource condition, int pageNum, int pageSize);

    SysResource selectByPrimaryKey(Long id);

    void save(SysResource sysResource);

    void delete(Long[] idList);

    List<Map> getResourceOptions();
}
